src/rgb/prefetch.rs

```diff
@@ -0,0 +1,239 @@
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+use amplify::{
+    confinement::Confined,
+    hex::{FromHex, ToHex},
+};
+use bdk::blockchain::EsploraBlockchain;
+use bech32::{decode, FromBase32};
+use bitcoin::{OutPoint, Script, Txid};
+use bitcoin_30::ScriptBuf;
+use bp::{LockTime, Outpoint, SeqNo, Tx, TxIn, TxOut, TxVer, Txid as BpTxid, VarIntArray, Witness};
+use rgb::{DeriveInfo, MiningStatus, RgbWallet, SpkDescriptor, Utxo};
+use rgbstd::containers::Contract;
+use std::{collections::BTreeMap, str::FromStr};
+use strict_encoding::StrictDeserialize;
+use wallet::onchain::ResolveTx;
+
+use super::resolvers::ExplorerResolver;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn prefetch_resolve_commit_utxo(contract: &str, explorer: &mut ExplorerResolver) {}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn prefetch_resolve_psbt_tx(asset_utxo: &str, explorer: &mut ExplorerResolver) {}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn prefetch_resolve_spend(
+    iface_index: u32,
+    wallet: RgbWallet,
+    explorer: &mut ExplorerResolver,
+) {
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn prefetch_resolve_watcher(
+    iface_index: u32,
+    explorer: &mut ExplorerResolver,
+    wallet: &mut RgbWallet,
+) {
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn prefetch_resolve_txs(txids: Vec<Txid>, explorer: &mut ExplorerResolver) {}
+
+#[cfg(target_arch = "wasm32")]
+pub async fn prefetch_resolve_commit_utxo(contract: &str, explorer: &mut ExplorerResolver) {
+    let esplora_client: EsploraBlockchain = EsploraBlockchain::new(&explorer.explorer_url, 100);
```

Comment:
Do you think we could use this method, and set the concurrency value to six? IIRC, browser pages can make up to six requests in parallel.
https://docs.rs/bdk/latest/bdk/blockchain/esplora/struct.EsploraBlockchain.html#method.with_concurrency

Also, let's move this to a function so any changes can be made in one place.